### DISCUSSION TITLE
relative path

### DISCUSSION THREAD

**Discussion by ariamVille** [opcodesio/log-viewer#388] [Q&A]
how do you configure with relative path? i set to 

'include_files' => [
    'storage/logs/*.log',
  ],
  
  but not working

**Comment by arukompas** [comment]
Hey @ariamVille , you can remove the "storage/logs" part because that is the default relative path it looks at. If you want to look outside the folder, then don't forget you can use "../" to go up a folder, for example. 

Hope this helps!

**Reply by ariamVille** [reply]
i see, by the way is there anything that needed to be configured? 

cause i can make it work only on local development but in other server environment it doesn't work 

here's my config so far

`<?php

$envPaths = explode(',', env('LOG_VIEWER_LOG_PATH_LIST', ''));
$includeFiles = array_map(function ($path) {
    return base_path($path);
}, $envPaths);

return [

    'enabled' => env('LOG_VIEWER_ENABLED', true),
    'api_only' => env('LOG_VIEWER_API_ONLY', false),
    'require_auth_in_production' => true,
    'route_domain' => null,
    'route_path' => 'log-viewer',
    'back_to_system_url' => config('app.url', null),
    'back_to_system_label' => null, // Displayed by default: "Back to {{ app.name }}"
    'timezone' => null,
    
    'middleware' => [
        'web',
        \Opcodes\LogViewer\Http\Middleware\AuthorizeLogViewer::class,
        'auth.admin'
    ],


    'api_middleware' => [
        \Opcodes\LogViewer\Http\Middleware\EnsureFrontendRequestsAreStateful::class,
        \Opcodes\LogViewer\Http\Middleware\AuthorizeLogViewer::class,
    ],

    'api_stateful_domains' => env('LOG_VIEWER_API_STATEFUL_DOMAINS') ? explode(',', env('LOG_VIEWER_API_STATEFUL_DOMAINS')) : null,

    'hosts' => [
        'local' => [
            'name' => ucfirst(env('APP_ENV', 'local')),
        ],

        // 'staging' => [
        //     'name' => 'Staging',
        //     'host' => 'https://staging.example.com/log-viewer',
        //     'auth' => [      // Example of HTTP Basic auth
        //         'username' => 'username',
        //         'password' => 'password',
        //     ],
        // ],
        //
        // 'production' => [
        //     'name' => 'Production',
        //     'host' => 'https://example.com/log-viewer',
        //     'auth' => [      // Example of Bearer token auth
        //         'token' => env('LOG_VIEWER_PRODUCTION_TOKEN'),
        //     ],
        //     'headers' => [
        //         'X-Foo' => 'Bar',
        //     ],
        // ],
    ],

    'include_files' => $includeFiles,


    'exclude_files' => [
        // 'my_secret.log'
    ],


    'hide_unknown_files' => true,

    'shorter_stack_trace_excludes' => [
        '/vendor/symfony/',
        '/vendor/laravel/framework/',
        '/vendor/barryvdh/laravel-debugbar/',
    ],


    'cache_driver' => env('LOG_VIEWER_CACHE_DRIVER', null),
    'cache_key_prefix' => 'lv',
    'lazy_scan_chunk_size_in_mb' => 50,
    'strip_extracted_context' => true,
];
`

**Reply by ariamVille** [reply]
> Hey @ariamVille , you can remove the "storage/logs" part because that is the default relative path it looks at. If you want to look outside the folder, then don't forget you can use "../" to go up a folder, for example.
> 
> Hope this helps!

even if i use this default *.log 
`LOG_VIEWER_LOG_PATH_LIST="*.log,../../test/*.log,../../public/*.log"`

it does not work with other server

**Reply by arukompas** [reply]
hey @ariamVille, calling `base_path` returns an absolute path, no longer a relative one. `base_path` also points to your app's base directory, where you usually find composer.json for example. And that's not where the logs are.

I see you want to have different includes in different environments (local, staging, production, etc), so here's how I would do it if I were you:

```php
<?php

return [
    // other settings

    'include_files' => !empty(env('LOG_VIEWER_LOG_PATH_LIST')) 
        ? explode(',', env('LOG_VIEWER_LOG_PATH_LIST')) 
        : [],

    // other settings
];
```

This way an environment value like `LOG_VIEWER_LOG_PATH_LIST="*.log,../../test/*.log,../../public/*.log"` will look for log files in these places:
- storage/logs/*.log
- storage/logs/../../test/*.log
- storage/logs/../../public/*.log
